### ISSUE TITLE
Docs: Incorrect part of the description regarding node:test setup in RuleTester

### ISSUE THREAD

**Issue by erm1116** [typescript-eslint/typescript-eslint#9822]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [X] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Suggested Changes

[In the document](https://typescript-eslint.io/packages/rule-tester/#node-built-in-test-runner), setup.js is written like below
```ts
// setup.js
import * as test from 'node:test';
import { RuleTester } from '@typescript-eslint/rule-tester';

RuleTester.afterAll = test.afterAll;
RuleTester.describe = test.describe;
RuleTester.it = test.it;
RuleTester.itOnly = test.it.only;
```

However, node test-runner doesn’t have `afterAll` function, it is `after`. ([Node.js Test runner doc](https://nodejs.org/api/test.html#afterfn-options))
So it should be fixed with `after`.

### Affected URL(s)

https://typescript-eslint.io/packages/rule-tester/#node-built-in-test-runner
